services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortHistorySpec.groovy

```diff
@@ -55,21 +59,22 @@ class PortHistorySpec extends HealthCheckSpecification {
         }
 
         then: "Port history is updated on the src switch"
-        def timestampAfterUp = System.currentTimeMillis()
-        with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterUp)) {
-            it.size() == 4
-            def antiFlapDeactivatedEvent = it.find { it.event == "ANTI_FLAP_DEACTIVATED" }
-            checkPortHistory(antiFlapDeactivatedEvent, isl.srcSwitch.dpId, isl.srcPort, "ANTI_FLAP_DEACTIVATED")
-            def portUpEvent = it.find { it.event == "PORT_UP" }
-            checkPortHistory(portUpEvent, isl.srcSwitch.dpId, isl.srcPort, "PORT_UP")
+        Long timestampAfterUp
+        Wrappers.wait(WAIT_OFFSET) {
+            timestampAfterUp = System.currentTimeMillis()
+            with(northboundV2.getPortHistory(isl.srcSwitch.dpId, isl.srcPort, timestampBefore, timestampAfterUp)
+                    .sort { a, b -> b.date <=> a.date }) {
+                checkPortHistory(it[0], isl.srcSwitch.dpId, isl.srcPort, "PORT_UP")
+                checkPortHistory(it[1], isl.srcSwitch.dpId, isl.srcPort, "ANTI_FLAP_DEACTIVATED")
```

Comment:
Antiflap can also be deactivated based on just port down (after cooldown). This is not tested here. I'm also wondering whether it makes sense just to add history verifications to AntiflapSpec?